backend/clap/src/main/java/zpi/ppea/clap/repository/PartialRatingRepository.java

```diff
@@ -31,7 +32,13 @@ public PartialRating upsertPartialRating(UpdatePartialRatingDto updatePartialRat
 
             PartialRatingRequest request = builder.build();
             log.info("Upserting partial rating:\n {}", request);
-            return dataStoreFutureStub.postPartialRating(request).get();
+            var response = dataStoreFutureStub.postPartialRating(request).get();
+
+            // We just detected race condition
+            if (response.hasError()) {
+                throw new RaceConditionException(null, authentication.getRefresh());
```

Comment:
@Remmo1 Can i throw it here, or should i catch it, do `Thread.currentThread().interrupt();` and only then rethrow it/

Reply:
I really don't have time today, merge it as it is. We have another ticket for it. Tomorrow we will disscus